background.ts

```diff
@@ -171,6 +171,15 @@ Electron.app.whenReady().then(async() => {
   try {
     const commandLineArgs = getCommandLineArgs();
 
+    // Normally `noModalDialogs` is set when we call `updateFromCommandLine(.., commandLineArgs)`
+    // But if there's an error either in that function, or before, we'll need to know if we should
+    // display the error in a modal-dialog or not. So check the current command-line arguments for that.
+    //
+    // It's very unlikely that a string option is set to this exact string though.
+    // `rdctl start --images.namespace --no-modal-dialogs`
+    // is syntactically correct, but unlikely (because why would someone create a
+    // containerd namespace called "--no-modal-dialogs"?
+    noModalDialogs = commandLineArgs.includes('--no-modal-dialogs');
```

Comment:
Is this really sufficient? Because `updateFromCommandLine` seems to support `--no-modal-dialogs false`, which would be incorrectly detected here as `true`.

I feel like the right way to implement this would be to have 2 functions: `parseCommandLine`, which would set global options, and also initialize a `Settings` object with everything specified on the commandline.

And later on `updateSettingsFromCommandLine` would get passed in this `Settings` object, and it would be processed just like a `PUT /settings` API call.

Reply:
The problem here is how do you detect `--no-modal-dialogs` set to true on a command-line that has other errors?

We can't make the default to never show a modal dialog unless it's specifically requested, for obvious reasons.

Maybe it should also be an environment variable, which we use, and can be overwritten with a command-line argument only when there are no errors processing the deployment files and the specified command-line.